src/main/java/com/elytradev/concrete/resgen/ConcreteResourcePack.java

```diff
@@ -212,7 +210,13 @@ private String generateItemModel(String name) {
 		if (Block.getBlockFromItem(itemFromLocation) != Blocks.AIR) {
 			if (itemFromLocation.getRegistryName().getResourcePath().equals(itemID)) {
 				try {
-					cache.put(name, IOUtils.toString(getInputStreamByName(name.replace("/item/", "/block/"))));
+					String blockName = name.replace("/item/", "/block/");
+					// Checks if we can use an existing block model or if we need to gen our own.
+					if ((boolean) hasResourceName.invoke(realResourcePack, blockName)) {
```

Comment:
No longer a wrapper; if a request falls through to this pack, it's safe to always generate.